client/express-checkout/index.js

```diff
@@ -185,41 +189,70 @@ jQuery( ( $ ) => {
 
 			const eceButton = wcpayECE.createButton( elements, {
 				buttonType: {
-					googlePay: wcpayExpressCheckoutParams.button.type,
-					applePay: wcpayExpressCheckoutParams.button.type,
+					googlePay: getExpressCheckoutData( 'button' ).type,
+					applePay: getExpressCheckoutData( 'button' ).type,
 				},
 			} );
 
 			wcpayECE.showButton( eceButton );
 
-			wcpayECE.attachButtonEventListeners( eceButton );
-
 			eceButton.on( 'click', function ( event ) {
 				const clickOptions = {
-					business: {
-						name: 'Mikes Bikes',
-					},
-					lineItems: [
-						{ name: 'Bike', amount: 200 },
-						{ name: 'Helmet', amount: 300 },
-					],
-					shippingAddressRequired: true,
-					shippingRates: [
-						{
-							id: '1',
-							amount: 500,
-							displayName: 'Standard Shipping',
-						},
-						{
-							id: '2',
-							amount: 1000,
-							displayName: 'Expedited Shipping',
-						},
-					],
+					lineItems: options.displayItems.map( ( i ) => ( {
+						...i,
+						name: i.label,
+					} ) ),
+					emailRequired: true,
+					shippingAddressRequired: options.requestShipping,
+					// FIXME: This is a total hack, we need some way to get the shipping information before rendering the button.
+					//        Possible to just send an empty address maybe? Just make up a free shipping thing since the shipping
+					//        address change event is sent as soon as the payment sheet pops up?
+					shippingRates: options.displayItems
+						.filter( ( i ) => i.label === 'Shipping' )
+						.map( ( i ) => ( {
+							id: `rate-${ i.label }`,
+							amount: i.amount,
+							displayName: i.label,
+						} ) ),
 				};
 				event.resolve( clickOptions );
 			} );
 
+			// FIXME: This handler is copied from ./event-handlers.js. We should re-use the same function here.
+			eceButton.on( 'shippingaddresschange', async ( event ) => {
+				const response = await api.expressCheckoutECECalculateShippingOptions(
+					normalizeShippingAddress( event.address )
+				);
+
+				if ( response.result === 'success' ) {
+					elements.update( { amount: response.total.amount } );
+					event.resolve( {
+						shippingRates: response.shipping_options,
+					} );
+				} else if ( response.result === 'fail' ) {
+					event.reject();
+				}
```

Comment:
It might be prudent to try to revert the changes here in some way, for example to reset the amount and/or selected shipping method, and something similar.

Reply:
Actually, I think the payment sheet does this automatically, so I don't think there's any action required here.